lib/subdir.am

```diff
@@ -2,7 +2,12 @@
 # libfrr
 #
 lib_LTLIBRARIES += lib/libfrr.la
-lib_libfrr_la_LDFLAGS = -version-info 0:0:0 -Xlinker -e_libfrr_version
+lib_libfrr_la_LDFLAGS = -version-info 0:0:0
+
+if LINUX
```

Comment:
It's nice for development to be able to easily double-check the library, and this did work on the BSDs last I checked.

Can you flip the check around and make it
```
if MACH
# nothing
else
lib_libfrr_la_LDFLAGS += -Xlinker -e_libfrr_version
endif
```

Reply:
Yes, sure although I'll have to add an AM_CONDITIONAL for MacOS.

But before doing that, I'd like to verify that this functionality still works on Linux. It doesn't for me at the moment:
```
$ /home/ruben/opt/lib/libfrr.so
Segmentation fault (core dumped)
```

Reply:
> Yes, sure although I'll have to add an AM_CONDITIONAL for MacOS.

Probably need that sooner or later anyway :)

> But before doing that, I'd like to verify that this functionality still works on Linux.

```
frr-master $ lib/.libs/libfrr.so
FRRouting 7.4-dev.
Copyright 1996-2005 Kunihiro Ishiguro, et al.
configured with:
    '-C' '--build=x86_64-linux-gnu' [...cut...]
```

> It doesn't for me at the moment:
> ```
> $ /home/ruben/opt/lib/libfrr.so
> Segmentation fault (core dumped)
> ```

Can you get me a backtrace for that?

Reply:
```
[ruben@fikkie frr]$ gdb ~/opt/lib/libfrr.so
GNU gdb (GDB) 9.1
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/ruben/opt/lib/libfrr.so...
(gdb) r
Starting program: /home/ruben/opt/lib/libfrr.so

Program received signal SIGSEGV, Segmentation fault.
0x00005555555a53f1 in _libfrr_version () at lib/libfrr.c:1167
1167		write(1, banner, sizeof(banner) - 1);
(gdb) thread apply all bt

Thread 1 (process 1084819):
#0  0x00005555555a53f1 in _libfrr_version () at lib/libfrr.c:1167
(gdb)
```

Reply:
Interesting... I have no idea wtf is going on there.  It seems it's not even calling `write()` since the location is still in `_libfrr_version`.  Are you using specific linker flags?  Which libc and version?

Worst case I'd just write this off as "works sometimes", it's not like we _need_ to be able to execute the library, just a gimmick ;)